### ISSUE TITLE
Numpy searializer issue.

### ISSUE THREAD

**Issue by jason-neal** [DanielAndreasen/SWEETer-Cat#105]
Some builds are failing due to.

```
 test_mpld3_post_request ____________________________
client = <FlaskClient <Flask 'app'>>
form_data = {'checkboxes': '', 'color': 'Blue', 'x': 'teff', 'x1': 'None', ...}
    def test_mpld3_post_request(client, form_data):
        data = {'x1': 'teff', 'x2': 'vt', 'y1': 'Vabs', 'y2': 'feh', 'z': 'logg'}
>       plot = client.post(url_for('mpld3_plot'), data=data, follow_redirects=False)
tests/test_plot.py:229: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/werkzeug/test.py:840: in post
    return self.open(*args, **kw)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/testing.py:127: in open
    follow_redirects=follow_redirects)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/werkzeug/test.py:803: in open
    response = self.run_wsgi_app(environ, buffered=buffered)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/werkzeug/test.py:716: in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/werkzeug/test.py:923: in run_wsgi_app
    app_rv = app(environ, start_response)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1997: in __call__
    return self.wsgi_app(environ, start_response)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1985: in wsgi_app
    response = self.handle_exception(e)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1540: in handle_exception
    reraise(exc_type, exc_value, tb)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/_compat.py:33: in reraise
    raise value
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1982: in wsgi_app
    response = self.full_dispatch_request()
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1614: in full_dispatch_request
    rv = self.handle_user_exception(e)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1517: in handle_user_exception
    reraise(exc_type, exc_value, tb)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/_compat.py:33: in reraise
    raise value
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1612: in full_dispatch_request
    rv = self.dispatch_request()
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1598: in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
app.py:63: in mpld3_plot
    return plot_page_mpld3(df, columns, request)
plot.py:127: in plot_page_mpld3
    plot = fig_to_html(fig)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/mpld3/_display.py:251: in fig_to_html
    figure_json=json.dumps(figure_json, cls=NumpyEncoder),
../../../../miniconda/envs/test-environment/lib/python3.6/json/__init__.py:238: in dumps
    **kw).encode(obj)
../../../../miniconda/envs/test-environment/lib/python3.6/json/encoder.py:199: in encode
    chunks = self.iterencode(o, _one_shot=True)
../../../../miniconda/envs/test-environment/lib/python3.6/json/encoder.py:257: in iterencode
    return _iterencode(o, 0)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/mpld3/_display.py:138: in default
    return json.JSONEncoder.default(self, obj)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <mpld3._display.NumpyEncoder object at 0x7f38fbb426d8>, o = array([1.])
    def default(self, o):
        """Implement this method in a subclass such that it returns
            a serializable object for ``o``, or calls the base implementation
            (to raise a ``TypeError``).
    
            For example, to support arbitrary iterators, you could
            implement default like this::
    
                def default(self, o):
                    try:
                        iterable = iter(o)
                    except TypeError:
                        pass
                    else:
                        return list(iterable)
                    # Let the base class default method raise the TypeError
                    return JSONEncoder.default(self, o)
    
            """
        raise TypeError("Object of type '%s' is not JSON serializable" %
>                       o.__class__.__name__)
E       TypeError: Object of type 'ndarray' is not JSON serializable
../../../../miniconda/envs/test-environment/lib/python3.6/json/encoder.py:180: TypeError
```

or 

```
test_mpld3_status_code ____________________________
client = <FlaskClient <Flask 'app'>>
    def test_mpld3_status_code(client):
>       c = client.get(url_for('mpld3_plot'))
tests/test_app.py:225: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/werkzeug/test.py:830: in get
    return self.open(*args, **kw)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/testing.py:127: in open
    follow_redirects=follow_redirects)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/werkzeug/test.py:803: in open
    response = self.run_wsgi_app(environ, buffered=buffered)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/werkzeug/test.py:716: in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/werkzeug/test.py:923: in run_wsgi_app
    app_rv = app(environ, start_response)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1997: in __call__
    return self.wsgi_app(environ, start_response)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1985: in wsgi_app
    response = self.handle_exception(e)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1540: in handle_exception
    reraise(exc_type, exc_value, tb)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/_compat.py:33: in reraise
    raise value
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1982: in wsgi_app
    response = self.full_dispatch_request()
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1614: in full_dispatch_request
    rv = self.handle_user_exception(e)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1517: in handle_user_exception
    reraise(exc_type, exc_value, tb)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/_compat.py:33: in reraise
    raise value
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1612: in full_dispatch_request
    rv = self.dispatch_request()
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/flask/app.py:1598: in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
app.py:63: in mpld3_plot
    return plot_page_mpld3(df, columns, request)
plot.py:127: in plot_page_mpld3
    plot = fig_to_html(fig)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/mpld3/_display.py:251: in fig_to_html
    figure_json=json.dumps(figure_json, cls=NumpyEncoder),
../../../../miniconda/envs/test-environment/lib/python3.6/json/__init__.py:238: in dumps
    **kw).encode(obj)
../../../../miniconda/envs/test-environment/lib/python3.6/json/encoder.py:199: in encode
    chunks = self.iterencode(o, _one_shot=True)
../../../../miniconda/envs/test-environment/lib/python3.6/json/encoder.py:257: in iterencode
    return _iterencode(o, 0)
../../../../miniconda/envs/test-environment/lib/python3.6/site-packages/mpld3/_display.py:138: in default
    return json.JSONEncoder.default(self, obj)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <mpld3._display.NumpyEncoder object at 0x7f38fab5c320>, o = array([1.])
    def default(self, o):
        """Implement this method in a subclass such that it returns
            a serializable object for ``o``, or calls the base implementation
            (to raise a ``TypeError``).
    
            For example, to support arbitrary iterators, you could
            implement default like this::
    
                def default(self, o):
                    try:
                        iterable = iter(o)
                    except TypeError:
                        pass
                    else:
                        return list(iterable)
                    # Let the base class default method raise the TypeError
                    return JSONEncoder.default(self, o)
    
            """
        raise TypeError("Object of type '%s' is not JSON serializable" %
>                       o.__class__.__name__)
E       TypeError: Object of type 'ndarray' is not JSON serializable
../../../../miniconda/envs/test-environment/lib/python3.6/json/encoder.py:180: TypeError
```

Similar to mpld3 [issues](https://github.com/mpld3/mpld3/issues) 434, 440.

I am working on a solution using  `df[col].tolist()` to remove numpy arrays when plotting.

**Comment by DanielAndreasen** [comment]
Did you manage to fix this?

**Comment by jason-neal** [comment]
I don't think so.
I tried merging this change into the failing PR #103  and it still failed with serializer issues in [build #371](https://travis-ci.org/DanielAndreasen/SWEETer-Cat/builds/337441829.) 
I have not looked at it since then.

The build itself is passing, and master is currently passing so I don't know if it will solve the issue even though it passes.

**Comment by DanielAndreasen** [comment]
Don't know either. I trust your judgement. As long as the master branch is healthy, I'm happy :)

**Comment by jason-neal** [comment]
I have got this issue fixed in https://github.com/mpld3/mpld3/pull/447 master. Mpld3 is not being maintained anymore so I do not know when/if this will be released on pypi.

**Comment by DanielAndreasen** [comment]
I didn't know that mpld3 is not maintained anymore. That's a shame. Not sure what the best solution is from here.

**Comment by jason-neal** [comment]
Neither did I until today and a note has been added if open an issue there.

It is apparently available if anybody wants to take over the project...

**Comment by jason-neal** [comment]
Should be temporarily solved with #129
